x-pack/solutions/security/packages/expandable-flyout/src/hooks/use_window_width.ts

```diff
@@ -9,14 +9,21 @@ import { useLayoutEffect, useState } from 'react';
 import { useDispatch } from '../store/redux';
 import { setDefaultWidthsAction } from '../store/actions';
 
-export const RIGHT_SECTION_MIN_WIDTH = 380;
 export const MIN_RESOLUTION_BREAKPOINT = 992;
-export const RIGHT_SECTION_MAX_WIDTH = 750;
 export const MAX_RESOLUTION_BREAKPOINT = 1920;
+export const BIG_RESOLUTION_BREAKPOINT = 2560;
+const FULL_WIDTH_BREAKPOINT = 1600;
+
+export const RIGHT_SECTION_MIN_WIDTH_OVERLAY_MODE = 380;
+export const RIGHT_SECTION_MAX_WIDTH_OVERLAY_MODE = 750;
 
+export const RIGHT_SECTION_MIN_WIDTH_PUSH_MODE = 380;
```

Comment:
consider grouping them into 2 objects for better readability

```
const Overlay = {
     RIGHT_SECTION_MIN_WIDTH = ...
     RIGHT_SECTION_Max_WIDTH...
     ....
}
```

Reply:
yup good idea! I went back and worth a ton on how to group them. It's way more complex that I though as some of these values are used in both push and overlay mode, for right and left sections...

I ended up refactoring this file quite a bit. I extracted some of the logic in 2 functions, which I think is a lot more clear as I could document them in details.

Let me know what you think of the new code!